docs/source/guides/upload.md

```diff
@@ -371,10 +377,91 @@ In addition to [`upload_file`] and [`upload_folder`], the following functions al
+getting an upload/push to fail at the end of the process or encountering a degraded experience, be it on hf.co or when working locally, can be very annoying.
+We gathered a list of tips and recommendations for structuring your repo.
+
+
+| Characteristic     | Recommended        | Tips                                     |
+| ----------------   | ------------------ | ---------------------------------------- |
+| Repo size          | -                  | contact us for large repos (TBs of data) |
+| Files per repo     | <100k              | merge data into fewer files              |
+| Entries per folder | <10k               | use subdirectories in repo               |
+| File size          | <5GB               | split data into chunked files            |
+| Commit size        | <100 files*        | upload files in multiple commits         |
+| Commits per repo   | -                  | upload multiple files per commit         |
+
+_* Not relevant when using `git` CLI directly_
+
+Please read the next section to understand better those limits and how to deal with them.
+
+### Hub repository size limitations
+
+What are we talking about when we say "large uploads", and what are their associated limitations? Large uploads can be
+very diverse, from repositories with a few huge files (e.g. model weights) to repositories with thousands of small files
+(e.g. an image dataset).
+
+Under the hood, the Hub uses Git to version the data, which has structural implications on what you can do in your repo.
+If your repo is crossing some of the numbers mentioned in the previous section, **we strongly encourage you to check out [`git-sizer`](https://github.com/github/git-sizer)**,
+which has very detailed documentation about the different factors that will impact your experience. Here is a TL;DR of factors to consider:
+
+- **Repository size**: The total size of the data you're planning to upload. There is no hard limit on a Hub repository size. However, if you plan to upload hundreds of GBs or even TBs of data, we would appreciate it if you could let us know in advance so we can better help you if you have any questions during the process. You can contact us at datasets@huggingface.co or on [our Discord](http://hf.co/join/discord).
+- **Number of files**:
+    - For optimal experience, we recommend keeping the total number of files under 100k. Try merging the data into fewer files if you have more.
+      For example, json files can be merged into a single jsonl file, or large datasets can be exported as Parquet files.
+    - The maximum number of files per folder cannot exceed 10k files per folder. A simple solution is to
+      create a repository structure that uses subdirectories. For example, a repo with 1k folders from `000/` to `999/`, each containing at most 1000 files, is already enough.
+- **File size**: In the case of uploading large files (e.g. model weights), we strongly recommend splitting them **into chunks of around 5GB each**.
```

Comment:
late to the party but i would have done larger, e.g. 20GB or at least 10GB (Cloudfront caches up to 30GB if i'm not mistaken)

@Pierrci @huggingface/moon-landing-back

Reply:
yes, but I don't really see the interest; as mentioned just below in the doc splitting into small chunks is better for uploading/downloading and retries, while I'm not sure there are a lot of advantages in doing chunks of 30GB?